lib/vnet/daemon/client_darwin.m

```diff
@@ -97,3 +72,90 @@ void OpenSystemSettingsLoginItems(void) {
     [SMAppService openSystemSettingsLoginItems];
   }
 }
+
+@interface VNEDaemonClient ()
+
+@property(nonatomic, strong, readwrite) NSXPCConnection *connection;
+@property(nonatomic, strong, readonly) NSString *bundlePath;
+
+@end
+
+@implementation VNEDaemonClient
+
+- (id)initWithBundlePath:(NSString *)bundlePath {
+  self = [super init];
+  if (self) {
```

Comment:
self can be null after [super init]? What happens then?  o.O

Reply:
I got it from here: https://www.mikeash.com/pyblog/the-how-and-why-of-cocoa-initializers.html

> And again for the sake of completeness, some may wonder why the if statement is there at all. The reason is because your superclass may fail to initialize (for example, if the parameters weren't consistent) and the standard way to indicate this is to release the object and return nil from the initializer. If you continue initializing your state after this happens, you will crash. The if statement lets you fail gracefully if your superclass failed.

In theory I could get rid of it since we don't even subclass anything specific, but this is addressed in the article as well:

> (As an aside, the standard pattern is not necessary if you subclass NSObject directly. This is because -[NSObject init] in documented to do nothing and always return self. However, using the principle that we should always write code that's robust to changes, in this case to changes in what class you inherit from, I very much recommend using the standard pattern even for direct NSObject subclasses.)

Reply:
Thanks for the explanation.